includes/chart.js

```diff
@@ -542,52 +545,85 @@ function getParentTransform(node) {
-                startRowIdNum -= 1;
-                destinationRowIdNum -= 1;
-            } else {
-                console.error("Invalid startShapeId or destinationShapeId");
-            }
+        drawConnection(event);
+    } else if (isDeletingConnection) {
+        deleteConnection(event);
+    }
+});
 
-            // Set the attributes with the updated values
-            line.setAttribute("data-start-row-id", startRowIdNum);
-            line.setAttribute("data-end-row-id", destinationRowIdNum);
-            line.setAttribute("id", "connector_" + startRowIdNum + "-" + destinationRowIdNum);
-            line.setAttribute("start-shape-id_", startShapeId);
-            line.setAttribute("end-shape-id_", destinationShapeId);
+function drawConnection(event) {
+    let destinationNode = event.target;
+    let destinationShapeId = destinationNode.id;
 
-            // Append the line to the edges group or svg container
-            connectionGroup.appendChild(line);
+    if (destinationNode.tagName !== 'ellipse' && destinationNode.tagName !== 'polygon' && destinationNode.tagName !== 'rect') {
+        return;
+    }
 
-            // Reset the drawing state
-            isDrawingConnection = false;
-            startShapeId = null;
-        }
+    console.log('Selected destination node:', destinationShapeId);
+    let startNode = document.getElementById(startShapeId);
+    let [startX, startY] = determineCenter(startNode);
+    let startTransform = getParentTransform(startNode);
+    let [endX, endY] = determineCenter(destinationNode);
+    let destinationTransform = getParentTransform(destinationNode);
+
+    // Create a line element and append it to the SVG
+    let line = document.createElementNS("http://www.w3.org/2000/svg", "line");
+    line.setAttribute("stroke", "green");
+    line.setAttribute("stroke-width", "5");
+    line.setAttribute("x1", startX + startTransform.translateX);
+    line.setAttribute("y1", startY + startTransform.translateY);
+    line.setAttribute("x2", endX + destinationTransform.translateX);
+    line.setAttribute("y2", endY + destinationTransform.translateY);
+
+    // Convert startShapeId and destinationShapeId to numbers
+    let startRowIdNum = parseInt(startShapeId, 10);
+    let destinationRowIdNum = parseInt(destinationShapeId, 10);
+
+    // Check if the conversion was successful
+    if (isNaN(startRowIdNum) || isNaN(destinationRowIdNum)) {
+        console.error("Invalid startShapeId or destinationShapeId");
     }
-});
+
+    // Set the attributes with the updated values
+    line.setAttribute("data-start-row-id", startRowIdNum);
+    line.setAttribute("data-end-row-id", destinationRowIdNum);
+    line.setAttribute("id", "connector_" + startRowIdNum + "-" + destinationRowIdNum);
+    line.setAttribute("start-shape-id_", startShapeId);
+    line.setAttribute("end-shape-id_", destinationShapeId);
+
+    console.log(line.getAttribute('id'));
```

Comment:
I would remove or comment (//) the console.log once the code is completed.

Reply:
Ah yes, that `console.log` is left from debugging, good catch!